common/persistence/cassandra/cassandraPersistence.go

```diff
@@ -827,6 +828,14 @@ workflow_state = ? ` +
 		`and type = ? ` +
 		`and task_id = ?`
 
+	templateRangeCompleteTaskQuery = `DELETE FROM tasks ` +
+		`WHERE domain_id = ? ` +
+		`AND task_list_name = ? ` +
+		`AND task_list_type = ? ` +
+		`AND type = ? ` +
+		`AND task_id >= ? ` +
+		`AND task_id <= ? `
```

Comment:
Exposing an API that deletes all rows > some_value is dangerous IMO. Imagine someone starting a whole bunch of workflows without starting workers and then abandoning it for a while. With this API, its possible for someone to issue a call to delete all rows > ackLevel. Agreed, a lot of things has to go wrong for this to happen - but I like the idea of forcing the callers to think about both lower/upper bound when issuing this call

If you feel strongly about this, we can chat offline.

Reply:
I don't think I understand the reasoning here.  We run AckManager for each tasklistMgr which moves the ack level only after all tasks in that range are completed.  So deleting all tasks below AckManager is perfectly safe.

Reply:
Discussed offline. Since RangeIDs (taskIDs) can jump in powers of two, there could be huge gaps in taskID namespace - so, exposing an API with lower/upper bound could result in more state at application level - doable but adds significant complexity. Conclusion is to expose an API that simply deletes rows < some_value but limit the number of rows deleted in a single db call.